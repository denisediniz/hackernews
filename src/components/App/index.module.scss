@import 'bulma/sass/utilities/_all.sass';
@import 'bulma/sass/base/helpers.sass';
@import 'bulma/sass/layout/section.sass';
@import 'bulma/sass/elements/container.sass';
@import 'bulma/sass/elements/content.sass';
