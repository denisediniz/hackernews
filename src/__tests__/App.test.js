import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import App, {
  updateSearchTopStoriesState,
  updateResultsStateOnDismiss
} from '../components/App';

describe('App', () => {
  it('renders without crashing', () => {
    shallow(<App />);
  });

  test('has a valid snapshot', () => {
    const component = renderer.create(<App />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('updateSearchTopStoriesState', () => {
  const firstReduxHits = [
    {
      title: 'Build Yourself a Redux',
      url: 'https://zapier.com/engineering/how-to-build-redux/',
      author: 'jdeal',
      num_comments: 155,
      points: 395,
      objectID: '14273549'
    }
  ];

  const secondReduxHits = [
    {
      title: 'Things to learn in React before using Redux',
      url: 'https://www.robinwieruch.de/learn-react-before-using-redux/',
      author: 'callumlocke',
      num_comments: 104,
      points: 370,
      objectID: '14811577'
    }
  ];

  const firstReactHits = [
    {
      title: 'Relicensing React, Jest, Flow, and Immutable.js',
      url: 'https://code.facebook.com/posts/300798627056246',
      author: 'dwwoelfel',
      num_comments: 498,
      points: 2280,
      objectID: '15316175'
    }
  ];

  const firstPrevState = {
    searchKey: 'redux',
    results: null
  };

  const secondPrevState = {
    searchKey: 'redux',
    results: {
      redux: {
        hits: [...firstReduxHits],
        page: 0
      }
    }
  };

  const thirdPrevState = {
    searchKey: 'react',
    results: {
      redux: {
        hits: [...firstReduxHits, ...secondReduxHits],
        page: 1
      }
    }
  };

  const firstReduxResult = {
    results: {
      redux: {
        hits: [...firstReduxHits],
        page: 0
      }
    },
    isLoading: false
  };

  const secondReduxResult = {
    results: {
      redux: {
        hits: [...firstReduxHits, ...secondReduxHits],
        page: 1
      }
    },
    isLoading: false
  };

  const firstMixedResult = {
    results: {
      redux: {
        hits: [...firstReduxHits, ...secondReduxHits],
        page: 1
      },
      react: {
        hits: [...firstReactHits],
        page: 0
      }
    },
    isLoading: false
  };

  it('updates the state correctly with null previous results', () => {
    const returnFn = updateSearchTopStoriesState(
      firstReduxHits,
      0
    )(firstPrevState);

    expect(returnFn).toEqual(firstReduxResult);
  });

  it('updates the state correctly with some previous results for the same term', () => {
    const returnFn = updateSearchTopStoriesState(
      secondReduxHits,
      1
    )(secondPrevState);

    expect(returnFn).toEqual(secondReduxResult);
  });

  it('updates the state correctly with some previous results for a different term', () => {
    const returnFn = updateSearchTopStoriesState(
      firstReactHits,
      0
    )(thirdPrevState);

    expect(returnFn).toEqual(firstMixedResult);
  });
});

describe('updateResultsStateOnDismiss', () => {
  const hits = [
    {
      title: 'Build Yourself a Redux',
      url: 'https://zapier.com/engineering/how-to-build-redux/',
      author: 'jdeal',
      num_comments: 155,
      points: 395,
      objectID: '14273549'
    },
    {
      title: 'Things to learn in React before using Redux',
      url: 'https://www.robinwieruch.de/learn-react-before-using-redux/',
      author: 'callumlocke',
      num_comments: 104,
      points: 370,
      objectID: '14811577'
    }
  ];

  const searchKey = 'redux';

  const prevState = {
    searchKey,
    results: {
      [searchKey]: {
        hits
      }
    }
  };

  it('matches if the hits array does not contain the removed element', () => {
    const dismissFn = updateResultsStateOnDismiss('14811577')(prevState);

    expect(dismissFn.results.redux.hits).toEqual(
      expect.not.objectContaining(hits[1])
    );
  });
});
