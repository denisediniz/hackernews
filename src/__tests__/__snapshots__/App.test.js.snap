// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App has a valid snapshot 1`] = `
<section
  className="section"
>
  <div
    className="container"
  >
    <div
      className="content"
    >
      <form
        onSubmit={[Function]}
      >
        <div
          className="field has-addons has-addons-centered"
        >
          <div
            className="control"
          >
            <input
              className="input"
              onChange={[Function]}
              type="text"
              value="redux"
            />
          </div>
          <div
            className="control"
          >
            <button
              className="button is-primary"
              onClick={[Function]}
              type="submit"
            >
              Search
            </button>
          </div>
        </div>
      </form>
    </div>
    <div
      className="content"
    >
      <div
        className="columns is-centered"
      >
        <div
          className="column is-narrow"
        >
          <p
            className="has-text-centered"
          >
            Order by
          </p>
          <div
            className="field has-addons has-addons-centered"
          >
            <p
              className="control"
            >
              <button
                className="button is-primary is-light"
                onClick={[Function]}
                type="button"
              >
                Title
              </button>
            </p>
            <p
              className="control"
            >
              <button
                className="button is-primary is-light"
                onClick={[Function]}
                type="button"
              >
                Author
              </button>
            </p>
            <p
              className="control"
            >
              <button
                className="button is-primary is-light"
                onClick={[Function]}
                type="button"
              >
                Comments
              </button>
            </p>
            <p
              className="control"
            >
              <button
                className="button is-primary is-light"
                onClick={[Function]}
                type="button"
              >
                Points
              </button>
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      className="content has-text-centered"
    >
      <button
        className="button is-primary is-loading"
      >
        Loading
      </button>
    </div>
  </div>
</section>
`;
